### ISSUE TITLE
Replacing an explicit declaration with an equivalent compiler-generated should not trigger RS0017

### ISSUE THREAD

**Issue by VladimirReshetnikov** [dotnet/roslyn-analyzers#5131]
See, for example, a declaration of an empty protected instance constructor in the class [`Microsoft.CodeAnalysis.Location`](http://source.roslyn.io/#Microsoft.CodeAnalysis/Diagnostic/Location.cs,15):

``` cs
    public abstract class Location
    {
        protected Location()
        {
        }

        // ...
```

This declaration is redundant. If you remove it from source, the compiler generates exactly the same by default. So, it does not affect a shape of public API in any way.

But when I tried to remove it, the build failed with an error:

```
error RS0017: Symbol 'Microsoft.CodeAnalysis.Location.Location()' is part of the declared API, but is either not public or could not be found
```

**EXPECTED:** This error does not occur.
